kahuna/public/js/components/gr-delete-image/gr-delete-image.js

```diff
@@ -3,23 +3,30 @@ import '../gr-confirm-delete/gr-confirm-delete';
 
 export const deleteImage = angular.module('gr.deleteImage', ['gr.confirmDelete']);
 
-deleteImage.controller('grDeleteImageCtrl', ['mediaApi', function (mediaApi) {
-    var ctrl = this;
+deleteImage.controller('grDeleteImageCtrl', [
+    '$rootScope', '$q', '$timeout', 'mediaApi',
+    function ($rootScope, $q, $timeout, mediaApi) {
+        var ctrl = this;
 
-    ctrl.delete = function () {
-        mediaApi.delete(ctrl.image)
-            .then((resp) => {
-                if (angular.isDefined(ctrl.onSuccess)) {
-                    ctrl.onSuccess(resp, ctrl.image);
-                }
-            })
-            .catch((err) => {
-                if (angular.isDefined(ctrl.onError)) {
-                    ctrl.onError(err, ctrl.image);
-                }
-            });
-    };
-}]);
+        ctrl.deleteImage = function (image) {
+            return mediaApi.delete(image)
+                .catch((err) => {
+                    $rootScope.$emit('image-delete-failure', err, image);
+                });
+        };
+
+        ctrl.delete = function () {
+            // HACK to wait for thrall to process the message so that when we
+            // poll the api, it will be up to date.
+            return $q.all(Array.from(ctrl.images.values()).map(image => ctrl.deleteImage(image)))
```

Comment:
Why not use the polling mechanic we have else where for checking when thralls done?
Or look into something like [`SSE`s](http://www.html5rocks.com/en/tutorials/eventsource/basics/)?